src/query.c

```diff
@@ -1057,15 +1057,46 @@ static IndexIterator *Query_EvalUnionNode(QueryEvalCtx *q, QueryNode *qn) {
 
 typedef IndexIterator **IndexIteratorArray;
 
+static void tag_strtofold(char *str, size_t *len, int caseSensitive) {
+  size_t origLen = *len;
+  char *origStr = str;
+  char *p = str;
+
+  while (*p) {
+    if (*p == '\\' && (ispunct(*(p+1)) || isspace(*(p+1)))) {
+      ++p;
+      --*len;
+    }
+    *str++ = *p++;
+  }
+  *str = '\0';
+
+  if (!caseSensitive) {
+    size_t newLen = unicode_tolower(origStr, origLen);
+    if (newLen && newLen < origLen) {
+      origStr[newLen] = '\0';
```

Comment:
Can also update `len` here and avoid calling `strlen` by the caller?

Reply:
Yes. Done.